### PR TITLE
Don't set the ad free cookie on reject all

### PR DESCRIPTION
**Pull request by emma-imber**
## What does this change?
- When a user rejects all, the ad free cookie will no longer be set
- Amends the consent Cypress tests to test the new logic
- Adds an article option to use the optout adtest

## Why?
This avoids us having to use a workaround for opt out advertising to remove the ad free cookie. Reject all is no longer an ad free experience with the introduction of opt out advertising, so this code change reflects that.

### CODE REVIEW

File: cypress/e2e/parallel-1/consent.cy.ts [hunk 1]
```diff
@@ -145,7 +135,7 @@ describe.skip('tcfv2 consent', () => {
 			.should('contain', 'Support the Guardian');
 	});
 
-	it(`Test ${path} shows ad slots when reconsented`, () => {
+	it.skip(`Test ${path} shows ad slots when reconsented`, () => {
```
[the remark this answers is not in the document]
Comment (emma-imber) [reply cypress/e2e/parallel-1/consent.cy.ts:138]: I think the tests are still valid and would be useful for testing the code, there's just a few different errors happening all at once that seem to affecting them that I don't think are in the scope of this PR.

**Comment by Jakeii** [comment]
I think we can get rid of the whole "ad free cookie reason" concept now couldn't we?

File: src/lib/manage-ad-free-cookie.ts [hunk 2]
```diff
@@ -11,7 +11,6 @@ const AD_FREE_USER_COOKIE = 'GU_AF1';
  * 3. The url parameter noadsaf is set so ads are temporarily disabled
  */
 enum AdFreeCookieReasons {
-	ConsentOptOut = 'consent_opt_out',
 	Subscriber = 'subscriber',
 	ForceAdFree = 'force_ad_free',
```
Comment (SiAdcock) [inline src/lib/manage-ad-free-cookie.ts:15]: If `ForceAdFree` can be [determined by `window.location.hash`](https://github.com/guardian/commercial/blob/4ccf6f96b86c05c11faa2baeb53ade7856ad1255/src/lib/user-features.ts#L52), do we need to store `AdFreeCookieReasons` in `localStorage` at all?

Comment (emma-imber) [reply src/lib/manage-ad-free-cookie.ts]: Good point both! I've updated the logic and tests. I'll test on CODE again tomorrow 🙂

File: src/commercial.consentless.ts [hunk 3]
```diff
@@ -11,24 +7,32 @@ import { initConsentless } from './lib/consentless/prepare-ootag';
...
 		setAdTestInLabelsCookie(),
 		initConsentless(consentState),
 		initExclusionSlot(),
 		initFixedSlots(),
 		initArticleInline(),
 		initLiveblogInline(),
-	]);
+	];
+
+	//this is added so that we can load the subscriber cookie for DCR pages and correctly hide ads
```
Comment (emma-imber) [inline src/commercial.consentless.ts:24]: I've added this change as at the moment we don't set the subscriber cookie when a user rejects all on DCR pages. As we've migrated more fronts to DCR this has become a problem, as a user needs to load a page via frontend to user user-features when they've rejected all. Adding this script here fixes the issue for now, but we should probably look to move this outside of the commercial bundle in the future.

**Comment by emma-imber** [comment]
After discussing the consent tests and their reasons for failing, we've decided to skip them for now, and add a ticket to the backlog to revisit them after getting some help from opt-out. Getting these tests working is still important, but not something that will be straightforward to do right now.

**Review by arelra — APPROVED** [review]
Great work @emma-imber on a very complex feature! Thanks for persisting through all the issues and spin off tickets. 🎉

File: src/lib/remove-targeted-ads-on-consent-change.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+			}
+		}
+	});
+};
+
+const removeTargetedAdsOnConsentChange = once(() =>
+	Promise.resolve(_removeTargetedAdsOnConsentChange()),
+);
+
+export const _ = { _removeTargetedAdsOnConsentChange };
```
[the remark this answers is not in the document]
Comment (emma-imber) [reply src/lib/remove-targeted-ads-on-consent-change.ts:23]: Yep good point! I'll amend this

File: src/lib/remove-targeted-ads-on-consent-change.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import { once } from 'lodash-es';
+import { removeSlots } from './remove-slots';
+
+/**
+ * If consent changes so that targeted advertising is disabled in tcfv2 regions,
+ * remove the ad slots from the page so that the reader no longer sees targeted advertising
+ */
+const _removeTargetedAdsOnConsentChange = (): void => {
+	onConsentChange((consent) => {
+		if (consent.framework === 'tcfv2') {
```
[the remark this answers is not in the document]
Comment (emma-imber) [reply src/lib/remove-targeted-ads-on-consent-change.ts:11]: Good shout, will do!

File: src/lib/remove-targeted-ads-on-consent-change.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { onConsentChange } from '@guardian/consent-management-platform';
+import { once } from 'lodash-es';
+import { removeSlots } from './remove-slots';
+
+/**
+ * If consent changes so that targeted advertising is disabled in tcfv2 regions,
+ * remove the ad slots from the page so that the reader no longer sees targeted advertising
+ */
+const _removeTargetedAdsOnConsentChange = (): void => {
```
[the remark this answers is not in the document]
Comment (emma-imber) [reply src/lib/remove-targeted-ads-on-consent-change.ts:9]: It's only removing ads from GAM that required consent, but isn't used to remove consentless ads. I can update the name to removeConsentedAds maybe to make that clearer

File: cypress/lib/util.ts [hunk 7]
```diff
@@ -38,7 +38,9 @@ export const getTestUrl = (
 		const builder = new URL(url);
 		builder.searchParams.append('adtest', adtest);
 		// force an invalid epic so it is not shown
-		builder.searchParams.append('force-epic', '9999:CONTROL');
+		if (adtest == 'fixed-puppies-ci') {
```
[the remark this answers is not in the document]
Comment (emma-imber) [reply cypress/lib/util.ts]: Good catch, I'll switch to ===

**Pull request merged**